README.md

```diff
@@ -25,13 +25,28 @@ The JVM calls sigaction, Golang requires that SA_ONSTACK flag be passed.
 Without this Golang will not be able to print the exception information eg. type, stack trace, line number.
 On Linux a solution is using LD_PRELOAD with a library that intercepts sigaction and adds the flag. (Code that does this: https://gist.github.com/timob/5d3032b54ed6ba2dc6de34b245c556c7)
 
+### MacOS AWT/Swing
+
+In order to run AWT/Swing on MacOS (only) is requires to run a platform specific Cocoa Run Loop code otherwise java gui becomes irresponsive.
+
+You would need to:
+
+1) Spin go routine in a *separate* thread from which is initialized JVM 
+2) In the application's main thread run `RunJVMGUILoop` function which configures and executes Cocoa run loop. 
+Running the loop is a blocking opperation for the current (main) thread. 
```

Comment:
If I am understanding this correctly the RunJVMLoop is the UI thread which handles events, like button clicks etc. So any non-UI code your App runs is in separate thread. Is that right?

This is similar to many UI toolkits, you can see how the UI loop works in SWT:
https://github.com/timob/swt/blob/master/example/example1.go#L97

Reply:
Separate thread and runloop is only requirement on mac and only if you use awt. Also some GUI libraries (do not have reference now) must not use this approach. this is reason for java launcher -XstartOnFirstThread [link](https://github.com/AdoptOpenJDK/openjdk-jdk/blob/704f784c88ee282837c980948167e741e7227f27/src/java.base/macosx/native/libjli/java_md_macosx.m#L866)

* Apple has excellent document describing the requirement and iterations:

[Section "Invoking the Java Virtual Machine From Native Code"](https://developer.apple.com/library/archive/technotes/tn2147/_index.html#//apple_ref/doc/uid/DTS10003827-CH1-INVOCATION)

* My code is based on [official Apple sample](https://developer.apple.com/library/archive/samplecode/simpleJavaLauncher/Introduction/Intro.html#//apple_ref/doc/uid/DTS10000688)

* OpenJDK java launcher is also using a very similar approach (https://github.com/AdoptOpenJDK/openjdk-jdk/blob/master/src/java.base/macosx/native/libjli/java_md_macosx.m#L354) 

* Good reference on the topic 

[OSX: JavaVM, AWT/Swing and possibly a deadlock](https://stackoverflow.com/questions/8750690/osx-javavm-awt-swing-and-possibly-a-deadlock/14859179)

A bit twisted setup on mac :)

Hope it helps as justification. My main purpose is to *allow* using AWT on mac by providing necessary code but it should be an optional.